### ISSUE TITLE
Switch to utf8mb4 on MySQL

### ISSUE THREAD

**Issue by kdambekalns** [neos/flow-development-collection#1268]
We use `utf8` for MySQL as character set, and `utf8m4_unicode_ci` as collation. This should rather be `utf8mb4` instead. Because:

- http://stackoverflow.com/questions/766809/
- http://dev.mysql.com/doc/refman/5.5/en/alter-table.html
- https://medium.com/@adamhooper/in-mysql-never-use-utf8-use-utf8mb4-11761243e434
- https://mathiasbynens.be/notes/mysql-utf8mb4
- https://florian.ec/articles/mysql-doctrine-utf8/

So I created these PRs:

- https://github.com/neos/flow-development-collection/pull/1267
- https://github.com/neos/neos-development-collection/pull/1991
- https://github.com/neos/neos-ui/pull/1758
- https://github.com/neos/party/pull/21
- https://github.com/neos/setup/pull/45
- https://github.com/neos/Neos.Demo/pull/39
- https://github.com/neos/BuildEssentials/pull/34

Basically setting up from scratch should work as usual. Custom migrations probably need to be adjusted (we can create a core migration, I think.) Using an existing database should work as well, the `./flow database:setcharset` can adjust things as needed.

This fixes https://github.com/neos/neos-development-collection/issues/869 as well.

**Comment by kdambekalns** [comment]
See https://discuss.neos.io/t/rfc-switch-to-utf8mb4-raise-db-requirements/3274

**Comment by kitsunet** [comment]
This was released with Flow 5.0, closing.
